test/e2e/play_kube_test.go

```diff
@@ -78,7 +78,19 @@ spec:
     image: {{ .Image }}
     name: {{ .Name }}
     imagePullPolicy: {{ .PullPolicy }}
-    resources: {}
+    {{- if or .CpuRequest .CpuLimit .MemoryRequest .MemoryLimit }}
+    resources:
+      {{- if or .CpuRequest .MemoryRequest }}
+      requests:
+        {{if .CpuRequest }}cpu: {{ .CpuRequest }}{{ end }}
+        {{if .MemoryRequest }}memory: {{ .MemoryRequest }}{{ end }}
+      {{- end }}
+      {{- if or .CpuLimit .MemoryLimit }}
+      limits:
+        {{if .CpuLimit }}cpu: {{ .CpuLimit }}{{ end }}
+        {{if .MemoryLimit }}memory: {{ .MemoryLimit }}{{ end }}
+      {{- end }}
+    {{- end }}
```

Comment:
deployment also need this, could you add the test for it?

Reply:
Certainly!